### PR TITLE
Initial pass at reinforcement learning plus training on Lichess data

### PR DESCRIPTION
**Pull request by owenjaques**
Four notebooks have been added which encompass the experiments @btfcopes and I have been running these past few weeks. In addition, two classes have been added in the neural_networks/chessbot directory which are used heavily in all the notebooks.

## New Classes
- **ChessBot:** Given a board evaluation model the class provides an interface to get the best _legal_ move from that model given a particular board state and returns it. 
- **ModelInput:** Translates boards from the python chess module into inputs that our neural network models expect.

## New Notebooks
- **reinforcement_learning.ipynb:** Runs games using the python chess module and trains the model after each game. Initially, the model is forced to make completely random moves but it is gradually allowed to make more moves (as opposed to forced random ones) as it learns more.
- **lichess_data.ipynb:** Downloads Lichess data then converts it into labelled training data that our models can then use to train. Currently labels the model inputs using the stockfish evaluation of the board state which is provided in the data.
- **data_make.ipynb:** Plays random games then labels and saves the data so a model can be trained on it later. Intended to save time as at the start of reinforcement learning the model learns by playing random games so this way we do not need to replay random games each time we train a new model.
- **RL_piece_val_reward.ipynb:** Another reinforcement learning experiment which labels the 'goodness' of a board according to the piece score that the bot currently has.

### CODE REVIEW

**Review by Kayisn — APPROVED** [review]
Thanks! I'll go through it again on Thursday after my test but looks good from a quick read through.
